include/experimental/__p1684_bits/mdarray.hpp

```diff
@@ -72,6 +72,8 @@ class mdarray {
   using container_type = Container;
   using mapping_type = typename layout_type::template mapping<extents_type>;
   using element_type = ElementType;
```

Comment:
Would you consider implementing the Mandates on `ElementType` ("a complete object type that is neither an abstract class type nor an array type")?

```suggestion
  static_assert(std::is_same_v<std::remove_cv_t<ElementType>, ElementType> &&
    ! std::is_abstract_v<ElementType> &&
    ! std::is_array_v<ElementType>);
  using element_type = ElementType;
```

I think we can even check whether `ElementType` is a complete type, by tricks like [the one explained here](https://devblogs.microsoft.com/oldnewthing/20190710-00/?p=102678).

Reply:
For sure but I am trying to do a couple incremental updates. So will do that in another one.